### PR TITLE
Implement user info controller

### PR DESCRIPTION
**Pull request by Xodoc**

### CODE REVIEW

File: InternshipBE/BL/Interfaces/IUserService.cs [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 {
-	public interface IUserService
-	{
-		Task<string> AuthenticateAsync(string email, string password);
-	}
+    public interface IUserService
+    {
+        Task<string> AuthenticateAsync(string email, string password);
+
+        Task<UserDTO> GetUserInfoAsync(string userId);
```
Comment (romanstat) [inline InternshipBE/BL/Interfaces/IUserService.cs]: aling, id

File: InternshipBE/WebApi/Startup.cs [hunk 2]
```diff
@@ -18,100 +18,106 @@
...
+            app.UseAuthentication();
+           
+            app.UseAuthorization();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapControllers();
+            });
+        }
+
```
Comment (romanstat) [inline InternshipBE/WebApi/Startup.cs]: empty line

File: InternshipBE/WebApi/Startup.cs [hunk 3]
```diff
@@ -18,100 +18,106 @@
...
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+                app.UseSwagger();
+                app.UseSwaggerUI(c => c.SwaggerEndpoint("/swagger/v1/swagger.json", "WebApi v1"));
+            }
+
+            app.UseGlobalExceptionMiddleware();
+
+            //app.UseHttpsRedirection();
```
Comment (romanstat) [inline InternshipBE/WebApi/Startup.cs]: remove comment

File: InternshipBE/WebApi/Startup.cs [hunk 4]
```diff
@@ -18,100 +18,106 @@
...
+                       ValidateAudience = true,
+                       ValidAudience = Configuration["Jwt:Audience"],
+                       ValidateLifetime = true,
+                       IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(Configuration["Jwt:Key"])),
+                       ValidateIssuerSigningKey = true,
+                   };
+               });
+
+            services.AddAutoMapper();
+        }
```
Comment (romanstat) [inline InternshipBE/WebApi/Startup.cs:101]: space

File: InternshipBE/BL/Services/UserService.cs [hunk 5]
```diff
@@ -12,50 +14,77 @@
...
+        }
+
+        public async Task<UserDTO> GetUserInfoAsync(string userId)
+        {
+            string userRole = string.Empty;
+
+            var user = await _userManager.FindByIdAsync(userId);
+            var roles = await _userManager.GetRolesAsync(user);
+
+            foreach (var role in roles)
```
Comment (romanstat) [inline InternshipBE/BL/Services/UserService.cs]: remove foreach

File: InternshipBE/WebApi/Controllers/UserController.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        private readonly IUserService _userService;
+
+        public UserController(IUserService userService)
+        {
+            _userService = userService;
+        }
+
+        [HttpGet("GetUserInfo")]
+        public async Task<IActionResult> GetUserInfo([FromQuery] string userId)
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/UserController.cs]: reimplement, remove userId from parameter

File: InternshipBE/BL/Services/UserService.cs [hunk 7]
```diff
@@ -12,50 +14,72 @@
...
+                return encodedJwt;
+            }
+
+            return string.Empty;
+        }
+
+        public async Task<UserDTO> GetUserInfoAsync(string id)
+        {
+            var user = await _userManager.FindByIdAsync(id);
+            var roles = await _userManager.GetRolesAsync(user);
```
Comment (romanstat) [inline InternshipBE/BL/Services/UserService.cs]: if user null throw exception

File: InternshipBE/WebApi/Controllers/UserController.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+        [HttpGet("GetUserInfo")]
+        public async Task<IActionResult> GetUserInfo([FromQuery] string id)
+        {
+            var user = await _userService.GetUserInfoAsync(id);
+
+            if (user == null)
+            {
+                return BadRequest("Invalid input data");
+            }
+
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/UserController.cs]: remove

File: InternshipBE/BL/Services/UserService.cs [hunk 9]
```diff
@@ -12,50 +14,72 @@
...
+
+            string userRole = roles[0];
+
+            if (user != null)
+            {
+                var userDTO = _mapper.Map<UserDTO>(user);
+
+                userDTO.RoleType = Enum.Parse<Shared.Enums.RoleType>(userRole);
```
Comment (romanstat) [inline InternshipBE/BL/Services/UserService.cs]: include namespace, remove Shared.Enums

File: InternshipBE/BL/Services/UserService.cs [hunk 10]
```diff
@@ -12,50 +15,77 @@
...
+
+            if (user != null)
+            {
+                var userDTO = _mapper.Map<UserDTO>(user);
+
+                userDTO.RoleType = Enum.Parse<RoleType>(userRole);
+
+                return userDTO;
+            }
+            return null;
```
Comment (romanstat) [inline InternshipBE/BL/Services/UserService.cs]: look 5 times at method and do refactor)

File: InternshipBE/WebApi/Controllers/UserController.cs [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        {
+            _userService = userService;
+        }
+
+        [HttpGet("GetUserInfo")]
+        public async Task<IActionResult> GetUserInfo()
+        {
+            var userName = User.Identity.Name;
+
+            var user = await _userService.GetUserByUserNameAsync(userName);
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/UserController.cs]: rename

File: InternshipBE/BL/Services/UserService.cs [hunk 12]
```diff
@@ -1,61 +1,109 @@
...
+            {
+                throw new ArgumentNullException(nameof(roles),"roles is null");
+            }
+            
+            string userRole = roles.SingleOrDefault(); 
+
+            return userRole;
+        }
+
+        private async Task<User> CheckPresenceUserAsync(string userName)
```
Comment (romanstat) [inline InternshipBE/BL/Services/UserService.cs]: rename

File: InternshipBE/WebApi/Controllers/UserController.cs [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    public class UserController : ControllerBase
+    {
+        private readonly IUserService _userService;
+
+        public UserController(IUserService userService)
+        {
+            _userService = userService;
+        }
+
+        [HttpGet("GetUserInfo")]
```
Comment (romanstat) [inline InternshipBE/WebApi/Controllers/UserController.cs]: getUserInfo

**Review by romanstat — APPROVED** [review]

**Review by vldmrklnk — APPROVED** [review]

**Review by Nikityanov — APPROVED** [review]

**Pull request merged**